### PR TITLE
[SYCL][Bindless] Enable SPIRV path for bindless_images

### PR DESCRIPTION
**Pull request by jsji**
This is 1st patch to enable SPIRV path for bindless_images so that we can start
generating SPIRV for testing purpose.

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Review by przemektmalon — APPROVED** [review]

File: sycl/test/extensions/bindless_images.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// RUN: %clangxx -S -emit-llvm -fsycl -fsycl-device-only -fsycl-targets=spir64-unknown-unknown %s -o - | FileCheck %s
+
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+// CHECK: spir_kernel void @_ZTSN4sycl3_V16detail19__pf_kernel_wrapperI10image_readEE
+// CHECK: tail call spir_func noundef <4 x float> @_Z17__spirv_ImageReadIDv4_fmiET_T0_T1_
```
Comment (MrSidims) [inline sycl/test/extensions/bindless_images.cpp:7]: That is a regular image read. Is it expected?

**Review by MrSidims — CHANGES_REQUESTED** [review]

File: sycl/test/extensions/bindless_images.cpp (hunk 1 above)
Comment (isaacault) [reply sycl/test/extensions/bindless_images.cpp:7]: Is it? I believe this is correct, defined [here](https://github.com/intel/llvm/blob/c1ef658791b0104780f715e783db923cbbd80004/libclc/ptx-nvidiacl/libspirv/images/image.cl#L1733)

**Review by isaacault — APPROVED** [review]

**Review by Seanst98 — APPROVED** [review]

File: sycl/test/extensions/bindless_images.cpp (hunk 1 above)
Comment (jsji) [reply sycl/test/extensions/bindless_images.cpp:7]: Thanks @isaacault . [here](https://github.com/intel/llvm/blob/c1ef658791b0104780f715e783db923cbbd80004/libclc/ptx-nvidiacl/libspirv/images/image.cl#L1733) implemented the spirv_ImageRead to call https://github.com/intel/llvm/blob/c1ef658791b0104780f715e783db923cbbd80004/libclc/ptx-nvidiacl/libspirv/images/image.cl#L1646C12-L1646C24 _nvvm_sul . We may need to do similar implementation somewhere, but that can be in a follow up PR? @MrSidims .

Comment (isaacault) [reply sycl/test/extensions/bindless_images.cpp]: As for your last commit, the Windows behaviour is expected - mangling happening [here](https://github.com/intel/llvm/blob/c1ef658791b0104780f715e783db923cbbd80004/libclc/ptx-nvidiacl/libspirv/images/image.cl#L20-L26).

**Review by isaacault — APPROVED** [review]

**Review by MrSidims — APPROVED** [review]

**Review by againull — APPROVED** [review]

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Can we get this merged? Thanks.

**Pull request merged**